build-system/server/variable-substitution-tester.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+let url;
+let variableSubstitution;
+let variables;
+
+function saveVariables(req, res) {
+  const requestVariables = {};
+  // For when a JSON is entered
+  if (req.query && Object.keys(req.query).length === 2) {
+    const entries = Object.entries(req.query);
+    try {
+      Object.assign(requestVariables, JSON.parse(entries[1][0]));
+    } catch (e) {
+      res.send(`<!doctype html>
+      <html>
+      <head>
+          <title>AMP Analytics</title>
+      </head> 
+      <body>
+      <p>Error:</p>
+      ${e}
+      </body>
+      </html>`);
+      return;
+    }
+  } else {
+    // Remove variables that don't have values
+    const keys = Object.keys(req.query);
+    for (let i = 0; i < keys.length; i++) {
+      if (req.query[keys[i]]) {
+        requestVariables[keys[i]] = req.query[keys[i]];
+      }
+    }
+  }
+  variables = requestVariables;
+  res.send(`<!doctype html>
```

Comment:
This could just be a .json endpoint instead of an .html, right?

Reply:
Will dig into it and see if it's possible.